packages/wallet-sdk/src/relay/WalletLinkRelay.ts

```diff
@@ -137,19 +134,27 @@ export class WalletLinkRelay extends WalletSDKRelayAbstract {
     const session = Session.load(this.storage) || new Session(this.storage).save();
 
     const connection = new WalletLinkConnection(
-      session.id,
-      session.key,
+      session,
       this.linkAPIUrl,
+      this.listener,
       this.diagnostic
     );
 
-    connection.setIncomingEventListener((m: ServerMessageEvent) => {
-      if (m.event === 'Web3Response') {
-        this.handleIncomingEvent(m);
-      }
+    const ui = this.options.uiConstructor({
+      linkAPIUrl: this.options.linkAPIUrl,
+      version: this.options.version,
+      darkMode: this.options.darkMode,
+      session,
     });
 
-    connection.setLinkedListener((linked: boolean) => {
+    connection.connect();
+
+    return { session, ui, connection };
+  }
```

Comment:
it's not new code
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/508c5eea-a144-4ebf-971d-fa2d44dfb4ac)